src/hpke/kem.rs

```diff
@@ -243,7 +243,7 @@ fn shared_secret_from_dh(alg: KEM, mut secret: Vec<u8>) -> Vec<u8> {
 /// [`ValidationError`](`HpkeError::ValidationError`) as described in
 /// [validation](#validation-of-inputs-and-outputs).
 pub fn DH(alg: KEM, sk: &PrivateKeyIn, pk: &PublicKeyIn) -> Result<SharedSecret, HpkeError> {
-    match crate::ecdh::derive(kem_to_named_group(alg).into(), pk, sk) {
+    match crate::ecdh::derive(kem_to_named_group(alg).try_into().unwrap(), pk, sk) {
```

Comment:
Using the KEM API here requires us to unwrap now. That's not great. But nothing to fix in here.
Can you file an issue to investigate whether to re-design the KEM - EC interactions/APIs?